diff --git a/configure.ac b/configure.ac
index 3030bad3d..525d582eb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -664,8 +664,9 @@ AC_ARG_ENABLE([hdf4-xdr],
               [AS_HELP_STRING([--enable-hdf4-xdr],
                               [Build xdr library from code in HDF4 source.
                                Not supported for 64 bit mode. [default="no"]])],
-              [enableval="yes"],[enableval="no"])
+              [],[enableval="no"])
 
+user_set_xdr="$enableval"
 case "$enableval" in
   yes)
     BUILD_XDR="yes"
@@ -693,7 +694,7 @@ if test "X$BUILD_XDR" != "Xyes"; then
     #include <rpc/types.h>
     #include <rpc/xdr.h>], [xdr_int],
                 [AC_MSG_RESULT([yes]); BUILD_XDR="no"],
-                [AC_MSG_RESULT([no]); BUILD_XDR="yes"])
+                [AC_MSG_RESULT([no]); if test "X$user_set_xdr" != "Xno"; then BUILD_XDR="yes"; fi])
 fi
 
 AM_CONDITIONAL([HDF_BUILD_XDR], [test "X$BUILD_XDR" = "Xyes"])
